Sample Applications/DataBindingDemo/AuctionItem.cs

```diff
@@ -45,6 +45,45 @@ protected void OnPropertyChanged(string name)
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
         }
 
+        public override string ToString()
+        {
+            // All important information conveyed visually in the AuctionItem must
+            // be conveyed programmatically. The use of border color is purely 
+            // decorative, and so does not need to be exposed programmatically. The
+            // use of the star however in the visuals is important, as is all the 
+            // text shown in the item. These data can be exposed programmatically 
+            // through the UI Automation (UIA) Name property of the item by returning 
+            // the desired string in ToString() here. That string will typically be
```

Comment:
Don't use ToString for this.  It breaks MVVM (by putting View information in the Model), and doesn't work in general.  For instance if the app had another view of an AuctionItem, or if the AuctionItem class wasn't owned by the app, you couldn't do this.

Instead set the Name property in the View, matching the visual described by that view as appropriate.  The mechanics to do this:  1) declare (or add to an existing) ItemContainerStyle for the ListBox, 2) set the AutomationProperties.Name attached property in that style, 3) use data-binding to make the name depend on the data item, 4) use multi-binding if the name depends on several properties, and a converter to piece them all together.

Sketch:

<ListBox.ItemContainerStyle>
  <Style TargetType="ListBoxItem">
     <Setter Property="AutomationProperties.Name">
         <Setter.Value>
             <MultiBinding Converter="{StaticResource.MyConverter}">
                <Binding Path="SpecialFeatures"/>
                <Binding Path="Description"/>
                <Binding Path="CurrentPrice"/>
 ///>

MyConverter.Convert(object[] values, ...)
{
   SpecialFeatures specialFeatures = ...convert values[0] to SpecialFeatures...;
   string description = values[1] as string;
   double currentPrice = values[2] as double;
   return   ... Guy's code ... ;
}